core/src/SYCL/Kokkos_SYCL_Instance.cpp

```diff
@@ -96,21 +114,9 @@ void SYCLInternal::initialize(const sycl::device& d) {
   const bool ok_init = nullptr == m_scratchSpace || nullptr == m_scratchFlags;
   const bool ok_dev  = true;
   if (ok_init && ok_dev) {
-    auto exception_handler = [](sycl::exception_list exceptions) {
-      bool asynchronous_error = false;
-      for (std::exception_ptr const& e : exceptions) {
-        try {
-          std::rethrow_exception(e);
-        } catch (sycl::exception const& e) {
-          std::cerr << e.what() << '\n';
-          asynchronous_error = true;
-        }
-      }
-      if (asynchronous_error)
-        Kokkos::Impl::throw_runtime_exception(
-            "There was an asynchronous SYCL error!\n");
-    };
-    m_queue.emplace(d, exception_handler);
+    m_queue = q;
+    all_queues.push_back(&m_queue);
```

Comment:
so we made the counter thread safe, i.e. we try to support multi threaded environments, but here we use push_back on a global? Lets protect this with a global mutex or so. Maybe have a static member function to insert into all_queues.